src/api/registrationApi.ts

```diff
@@ -71,11 +71,27 @@ export const deleteTicketMessage = async (ticketId: string, messageId: string) =
   });
 };
 
-export const createTicket = async (registrationId: string, type: TicketType, returnCreatedTicket = false) => {
+export const createTicket = async (
+  registrationId: string,
+  type: TicketType,
+  returnCreatedTicket = false,
+  message?: string
+) => {
   const createTicketResponse = await authenticatedApiRequest<null>({
     url: `${registrationId}/ticket`,
     method: 'POST',
-    data: { type },
+    data: {
+      type: type,
+      ...(message &&
+        message.length > 0 && {
+          messages: [
+            {
+              type: 'Message',
+              text: message,
+            },
+          ],
+        }),
```

Comment:
Måtte studere denne en stund for å skjønne hva den gjør, kan det forenkles så det blir mer intuitivt, feks noe sånt?

```suggestion
      message: message
        ? {
            messages: [{ type: 'Message', text: message }],
          }
        : undefined,
```

Reply:
Nei, for problemet er at hvis message er tom så bør jeg sende inn samme type objekt som før (altså uten message i det hele tatt), hvis ikke lagrer API'et en tom message og vi lister den ut i frontend. Backend har planer om å begynne å returnere error om de får en tom message. Derfor må jeg sjekke om vi har message før jeg legger til den propertien på objektet.

Reply:
Ah, det var litt uheldig da, men skjønner.

Om man vil fjerne message helt kan man også gjøre noe ala dette:
```suggestion
    data: message ? { type, messages: [{ type: 'Message', text: message }] } : { type },
```

Grunne til at jeg sier det er at jeg ikke forstår hva som egentlig skjer, siden det ligger en spread med en bool sjekk inni. Når message-sjekken er true lages det et objekt som spreades inn i `data` som er greit nok, men ville egentlig forventa en feil når `message` mangler. For da vil vi spreade inn enten `false` eller `undefined` som jeg ville trodd førte til en feil 🤔